homeassistant/components/nextbus/config_flow.py

```diff
@@ -0,0 +1,264 @@
+                for key, value in options.items()
+            ],
+            mode=SelectSelectorMode.DROPDOWN,
+        )
+    )
+
+
+class NextBusFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle Nextbus configuration."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize NextBus config flow."""
+        self.nextbus_config: dict[str, str] = {}
+        self._client = NextBusClient(output_format="json")
+        self._agency_tags: dict[str, str] = {}
+        self._route_tags: dict[str, str] = {}
+        self._stop_tags: dict[str, str] = {}
+        _LOGGER.info("Init new config flow")
+
+    async def _update_agency_config_options(self) -> None:
+        def job():
+            self._agency_tags = {
+                a["tag"]: a["title"] for a in self._client.get_agency_list()["agency"]
+            }
+
+        await self.hass.async_add_executor_job(job)
+
+    async def _update_route_config_options(self, agency_tag: str) -> None:
+        def job():
+            self._route_tags = {
+                a["tag"]: a["title"]
+                for a in self._client.get_route_list(agency_tag)["route"]
+            }
+
+        await self.hass.async_add_executor_job(job)
+
+    async def _update_stop_config_options(
+        self, agency_tag: str, route_tag: str
+    ) -> None:
+        def job():
+            self._stop_tags = {
+                a["tag"]: a["title"]
+                for a in self._client.get_route_config(
+                    route_tag,
+                    agency_tag,
+                )[
+                    "route"
+                ]["stop"]
+            }
+
+        await self.hass.async_add_executor_job(job)
+
+    async def async_step_import(
+        self, config_input: UserConfig | None = None
+    ) -> FlowResult:
+        """Handle import of config."""
+        if not config_input:
+            return self.async_abort(
+                reason="Import failed due to no data to migrate",
+            )
+
+        await self._update_agency_config_options()
+        self.nextbus_config[CONF_AGENCY] = config_input[CONF_AGENCY]
+
+        await self._update_route_config_options(config_input[CONF_AGENCY])
+        self.nextbus_config[CONF_ROUTE] = config_input[CONF_ROUTE]
+
+        await self._update_stop_config_options(
+            config_input[CONF_AGENCY], config_input[CONF_ROUTE]
+        )
+        self.nextbus_config[CONF_STOP] = config_input[CONF_STOP]
+
+        self.nextbus_config[CONF_NAME] = (
+            config_input.get(CONF_NAME)
+            or f"{config_input[CONF_AGENCY]} {config_input[CONF_ROUTE]}"
+        )
+
+        errors = self._validate_config()
+        if errors:
+            _LOGGER.error(errors)
+            return await self.async_step_user()
+
+        return self.async_create_entry(
+            title=self.nextbus_config[CONF_NAME],
+            data=self.nextbus_config,
+        )
```

Comment:
Better to be explicit
```suggestion
        return self.async_create_entry(
            title=config_input.get(CONF_NAME) or f"{agency_tags[agency]} {route_tags[route]} {stop_tags[stop]}",
            data={CONF_AGENCY=agency, CONF_ROUTE=route, CONF_STOP=stop},
        )
```